chia/util/streamable.py

```diff
@@ -67,28 +67,35 @@ class Field:
     name: str
     type: Type[object]
     has_default: bool
+    stream_function: StreamFunctionType
+    parse_function: ParseFunctionType
+    convert_function: ConvertFunctionType
+    post_init_function: ConvertFunctionType
 
 
-# Caches to store the fields and (de)serialization methods for all available streamable classes.
-FIELDS_FOR_STREAMABLE_CLASS: Dict[Type[object], Tuple[Field, ...]] = {}
-STREAM_FUNCTIONS_FOR_STREAMABLE_CLASS: Dict[Type[object], List[StreamFunctionType]] = {}
-PARSE_FUNCTIONS_FOR_STREAMABLE_CLASS: Dict[Type[object], List[ParseFunctionType]] = {}
-CONVERT_FUNCTIONS_FOR_STREAMABLE_CLASS: Dict[Type[object], List[ConvertFunctionType]] = {}
-POST_INIT_FUNCTIONS_FOR_STREAMABLE_CLASS: Dict[Type[object], List[ConvertFunctionType]] = {}
+StreamableFields = Tuple[Field, ...]
 
 
-def create_fields_cache(cls: Type[object]) -> Tuple[Field, ...]:
+def create_fields(cls: Type[object]) -> StreamableFields:
     hints = get_type_hints(cls)
-    fields = tuple(
-        Field(
-            name=field.name,
-            type=hints.get(field.name, None),
-            has_default=field.default is not dataclasses.MISSING or field.default_factory is not dataclasses.MISSING,
+    fields = []
+    for field in dataclasses.fields(cls):
+        hint = hints.get(field.name, None)
+        assert hint is not None
```

Comment:
?  At which point it can be inlined in the `Field` call like everything else and the list can be defined via comprehension.  Alternatively, a real message could be provided (with a 'regular' exception?) that enhances debuggability by reporting the the `cls` and the field name with some words.

```suggestion
        hint = hints[field.name]
```

Reply:
Added but inlining it requires `hints[field.name]` in multiple places since `hint` is used also for the functions below.